tests/epics/pmac/test_pmac_trajectory.py

```diff
@@ -0,0 +1,158 @@
+from unittest.mock import call
+
+import numpy as np
+import pytest
+from scanspec.specs import Fly, Line
+
+from ophyd_async.core import init_devices
+from ophyd_async.epics.motor import Motor
+from ophyd_async.epics.pmac import PmacIO
+from ophyd_async.epics.pmac._pmac_trajectory import (
+    PmacTrajectoryTriggerLogic,  # noqa: PLC2701
+)
+from ophyd_async.epics.pmac._utils import (
+    _PmacMotorInfo,  # noqa: PLC2701
+    _Trajectory,  # noqa: PLC2701
+)
+from ophyd_async.testing import get_mock, set_mock_value
+
+
+@pytest.fixture
+async def sim_motors():
+    async with init_devices(mock=True):
+        sim_x_motor = Motor("BLxxI-MO-STAGE-01:X")
+        sim_y_motor = Motor("BLxxI-MO-STAGE-01:Y")
+        sim_pmac = PmacIO(
+            prefix="Test_PMAC",
+            raw_motors=[sim_x_motor, sim_y_motor],
+            coord_nums=[1],
+        )
+
+    pmac_x = sim_pmac.assignment[sim_pmac.motor_assignment_index[sim_x_motor]]
+    pmac_y = sim_pmac.assignment[sim_pmac.motor_assignment_index[sim_y_motor]]
+    set_mock_value(pmac_x.cs_port, "CS1")
+    set_mock_value(pmac_x.cs_number, 1)
+    set_mock_value(pmac_x.cs_axis_letter, "X")
+    set_mock_value(pmac_y.cs_port, "CS1")
+    set_mock_value(pmac_y.cs_number, 1)
+    set_mock_value(pmac_y.cs_axis_letter, "Y")
+    set_mock_value(sim_x_motor.acceleration_time, 0.5)
+    set_mock_value(sim_x_motor.velocity, 1)
+    set_mock_value(sim_x_motor.max_velocity, 5)
+    set_mock_value(sim_y_motor.acceleration_time, 0.5)
+    set_mock_value(sim_y_motor.velocity, 1)
+    set_mock_value(sim_y_motor.max_velocity, 10)
+    yield (sim_pmac, sim_x_motor, sim_y_motor)
+
+
+async def test_pmac_prepare(sim_motors: tuple[PmacIO, Motor, Motor]):
+    pmacIO, sim_x_motor, sim_y_motor = sim_motors
+    spec = Fly(2.0 @ Line(sim_x_motor, 1, 5, 2))
+    pmac_trajectory = PmacTrajectoryTriggerLogic(pmacIO)
+    await pmac_trajectory.prepare(spec)
+
+    assert await pmacIO.coord[1].cs_axis_setpoint[7].get_value() == -1.2
+
+    assert await pmacIO.trajectory.positions[7].get_value() == pytest.approx(
+        [-1.0, 1.0, 3.0, 5.0, 7.0, 7.2]
+    )
+    assert pmac_trajectory.scantime == 4400000
```

Comment:
Should: Given we don't use sim_y_motor here, but the fixture returns it, I would use an underscore for clarity:

```suggestion

async def test_pmac_prepare(sim_motors: tuple[PmacIO, Motor, Motor]):
    pmacIO, sim_x_motor, _ = sim_motors
    spec = Fly(2.0 @ Line(sim_x_motor, 1, 5, 2))
    pmac_trajectory = PmacTrajectoryTriggerLogic(pmacIO)
    await pmac_trajectory.prepare(spec)

    assert await pmacIO.coord[1].cs_axis_setpoint[7].get_value() == -1.2

    assert await pmacIO.trajectory.positions[7].get_value() == pytest.approx(
        [-1.0, 1.0, 3.0, 5.0, 7.0, 7.2]
    )
    assert pmac_trajectory.scantime == 4400000
```
Similarily with other trajectory tests

Reply:
Done!